machine/machine.go

```diff
@@ -122,6 +124,44 @@ func GetClockSpeed(procInfo []byte) (uint64, error) {
 	return uint64(speed * 1000), nil
 }
 
+func GetVMStats(vmStatMetrics *string) (map[string]int, error) {
+	vmstatRegexp, err := regexp.Compile(*vmStatMetrics)
+	if err != nil {
+		return nil, fmt.Errorf("failed to compile vmstat regexp: %v", err)
+	}
+
+	stats := make(map[string]int)
+	file, err := os.Open(vmstatFile)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open %s", vmstatFile)
+	}
+	scanner := bufio.NewScanner(file)
+
+	for scanner.Scan() {
+		splittedLine := strings.SplitN(scanner.Text(), " ", 3)
+		if len(splittedLine) < 2 {
+			klog.Info("cannot parse vmstat's metric. Skipping")
+			continue
+		}
+		match := vmstatRegexp.MatchString(splittedLine[0])
+		if !match {
+			continue
+		}
+		value, err := strconv.Atoi(splittedLine[1])
+		if err != nil {
+			klog.Infof("cannot parse vmstat's %s. Skipping", splittedLine[0])
```

Comment:
If this error appears then it is an issue with `splittedLine[0]`, to easier debugging potential issue I think that it is worth to add information about `splittedLine[0]` .

Reply:
sorry, I was thinking about `splittedLine[1]` , it still needs to be addressed